File: 53-maximum-subarray/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        n= len(nums)
        best=[0]*n
        
        if n ==1:
            return nums[0]
        
        best[n-1] = nums[n-1]
        
        for i in  reversed(range(len(nums)-1)):
            best[i] = max(nums[i], nums[i]+best[i+1])
            
        maxi=best[0]
        
        for i in best:
            maxi = max(maxi, i)
            
        return maxi
            

File: Medium/53-maximum-subarray/53-maximum-subarray.cs
public class Solution {
    public int MaxSubArray(int[] nums) {
       
        int size = nums.Length;
        int[] arr = new int[size];
        
        arr[0] = nums[0];
        
        for ( int i =1; i< size; i++) {
            if (arr[i-1] + nums[i] > nums[i]){
                arr[i] = arr[i-1] + nums[i];
            } else {
                arr[i] = nums[i];
            }
        }
        
        int max = arr[0];
        foreach (int a in arr) {
            max = max > a ? max: a;
        }
        
        return max;
    }
}